cmd/auth/logout.go

```diff
@@ -0,0 +1,139 @@
+package auth
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io/fs"
+
+	"github.com/databricks/cli/libs/auth"
+	"github.com/databricks/cli/libs/auth/cache"
+	"github.com/databricks/cli/libs/cmdio"
+	"github.com/databricks/cli/libs/databrickscfg"
+	"github.com/databricks/cli/libs/databrickscfg/profile"
+	"github.com/databricks/databricks-sdk-go/config"
+	"github.com/spf13/cobra"
+)
+
+type LogoutSession struct {
+	Profile        string
+	File           config.File
+	PersistentAuth *auth.PersistentAuth
+}
+
+func (l *LogoutSession) load(ctx context.Context, profileName string, persistentAuth *auth.PersistentAuth) error {
+	l.Profile = profileName
+	l.PersistentAuth = persistentAuth
+	iniFile, err := profile.DefaultProfiler.Get(ctx)
+	if errors.Is(err, fs.ErrNotExist) {
+		return err
+	} else if err != nil {
+		return fmt.Errorf("cannot parse config file: %w", err)
+	}
+	l.File = *iniFile
+	if err := l.setHostAndAccountIdFromProfile(); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (l *LogoutSession) setHostAndAccountIdFromProfile() error {
+	sectionMap, err := l.getConfigSectionMap()
+	if err != nil {
+		return err
+	}
+	if sectionMap["host"] == "" {
+		return fmt.Errorf("no host configured for profile %s", l.Profile)
+	}
+	l.PersistentAuth.Host = sectionMap["host"]
+	l.PersistentAuth.AccountID = sectionMap["account_id"]
+	return nil
+}
+
+func (l *LogoutSession) getConfigSectionMap() (map[string]string, error) {
+	section, err := l.File.GetSection(l.Profile)
+	if err != nil {
+		return map[string]string{}, fmt.Errorf("profile does not exist in config file: %w", err)
+	}
+	return section.KeysHash(), nil
+}
+
+// clear token from ~/.databricks/token-cache.json
+func (l *LogoutSession) clearTokenCache(ctx context.Context) error {
+	return l.PersistentAuth.ClearToken(ctx)
+}
+
+// Overrewrite profile to .databrickscfg without fields marked as sensitive
+// Other attributes are preserved.
+func (l *LogoutSession) clearConfigFile(ctx context.Context, sectionMap map[string]string) error {
+	return databrickscfg.SaveToProfile(ctx, &config.Config{
+		ConfigFile:           l.File.Path(),
+		Profile:              l.Profile,
+		Host:                 sectionMap["host"],
+		ClusterID:            sectionMap["cluster_id"],
+		WarehouseID:          sectionMap["warehouse_id"],
+		ServerlessComputeID:  sectionMap["serverless_compute_id"],
+		AccountID:            sectionMap["account_id"],
+		Username:             sectionMap["username"],
+		GoogleServiceAccount: sectionMap["google_service_account"],
+		AzureResourceID:      sectionMap["azure_workspace_resource_id"],
+		AzureClientID:        sectionMap["azure_client_id"],
+		AzureTenantID:        sectionMap["azure_tenant_id"],
+		AzureEnvironment:     sectionMap["azure_environment"],
+		AzureLoginAppID:      sectionMap["azure_login_app_id"],
+		ClientID:             sectionMap["client_id"],
+		AuthType:             sectionMap["auth_type"],
+	})
+}
```

Comment:
IMO, the inclusion of this is to be discussed.

Reply:
My line of thinking was that a User might not want to remove their profiles entirely when they potentially think they are just logging out, since they might just do it for security reasons. So eventual cluster ID etc are not needed to be reconfigured on next login. 

And that it might be intuitive for a user who dont know the implementation details of the CLI; that it also would remove PAT token if it exists.

However I will of course respect and adapt to your opinion in this matter!